wgengine/magicsock: don't change DERP home when not connected to control

andrew-d opened the pull request:
This pretty much always results in an outage because peers won't discover our new home region and thus won't be able to establish connectivity.

Updates https://github.com/tailscale/corp/issues/18095


Change-Id: Ic0d09133f198b528dd40c6383b16d7663d9d37a7

raggi reviewed (approved):
Change makes sense, I wish we had easier to write independent coverage for this

bradfitz commented on wgengine/magicsock/magicsock.go:
```diff
@@ -682,7 +683,20 @@ func (c *Conn) updateNetInfo(ctx context.Context) (*netcheck.Report, error) {
+	// discover that change.
+	//
+	// See https://github.com/tailscale/corp/issues/18095
+	connectedToControl := health.GetInPollNetMap()
```
hah, I forgot about that. Nice.

andrew-d commented:
> Change makes sense, I wish we had easier to write independent coverage for this

I just refactored this bit out into a new function and added a test for it; PTAL?
